rama-http/src/headers/x_robots_tag_components/valid_date.rs

```diff
@@ -0,0 +1,63 @@
+use chrono::{DateTime, Utc};
+use rama_core::error::{ErrorContext, OpaqueError};
+use std::ops::Deref;
+use std::str::FromStr;
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub(super) struct ValidDate(DateTime<Utc>);
+
+impl ValidDate {
+    pub(super) fn new(date: DateTime<Utc>) -> Result<Self, OpaqueError> {
+        Ok(Self(date))
+    }
+}
+
+impl Deref for ValidDate {
+    type Target = DateTime<Utc>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl From<ValidDate> for DateTime<Utc> {
+    fn from(value: ValidDate) -> Self {
+        value.0
+    }
+}
+
+impl TryFrom<DateTime<Utc>> for ValidDate {
```

Comment:
1. if you have this, why still a constructor? Trait impls are always public (if the trait is public), so also makes the `pub(super)` restriction a bit silly
2. why is this fallible? Am I missing something? Can this not be just a normal `impl From<DateTime<Utc>>`?